### PR TITLE
[RW-5665][risk=no] Puppeteer snowman menu change

### PR DESCRIPTION
**Pull request by aweng98**
`snowman` is UI component in AoU workbench UI, thus `ellipsis-menu` is renamed to `snowman-menu`. Changed phrase `menuitem

Replaced occurrences of `ellipsis` string with `snowman`.

### CODE REVIEW

File: e2e/app/component/card-base.ts [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+import {Option} from 'app/text-labels';
+import Container from 'app/container';
+import SnowmanMenu from './snowman-menu';
 
 
-export default abstract class CardBase {
+export default abstract class CardBase extends Container {
 
-  ellipsisXpath = './/clr-icon[@shape="ellipsis-vertical"]';
+  private snowmanIconXpath = './/clr-icon[@shape="ellipsis-vertical"]';
```
Comment (als364) [inline e2e/app/component/card-base.ts:9]: How come this string is duplicated everywhere instead of living in `snowman-menu.ts`?

File: e2e/app/page/workspace-base.ts [hunk 2]
```diff
@@ -222,20 +222,22 @@ export default abstract class WorkspaceBase extends AuthenticatedPage {
...
-   * @param {EllipsisMenuAction} menuOption
+   * Select Workspace action snowman menu option.
+   * @param {Option} option
    * @param opts
    */
-  async selectWorkspaceAction(menuOption: EllipsisMenuAction, opts?: { waitForNav: false }): Promise<void> {
-    const ellipsisMenu = new EllipsisMenu(this.page, './/*[@data-test-id="workspace-menu-button"]');
-    return ellipsisMenu.clickAction(menuOption, opts);
+  async selectWorkspaceAction(option: Option, opts?: { waitForNav: false }): Promise<void> {
+    const iconXpath = './/*[@data-test-id="workspace-menu-button"]';
```
Comment (als364) [inline e2e/app/page/workspace-base.ts:230]: how come you're accessing this this way instead of the usual way of looking for the ellipsis icon?

File: e2e/tests/conceptsets/conceptset-edit.spec.ts [hunk 3]
```diff
@@ -19,13 +19,13 @@ describe('Editing and Copying Concept Sets', () => {
...
    * Test:
-   * - Create new workspace.
+   * - Find an existing workspace.
    * - Create first new Concept Set in Procedure domain.
    * - Create second new Concept Set in Procedure domain. Add to first Concept Set.
    */
   test('Add to existing Concept Set in same workspace', async () => {
     // Create a workspace
-    const workspaceName = await findWorkspace(page, {create: true}).then(card => card.clickWorkspaceName());
+    const workspaceName = await findWorkspace(page).then(card => card.clickWorkspaceName());
```
Comment (als364) [inline e2e/tests/conceptsets/conceptset-edit.spec.ts:28]: This doesn't seem related to the ellipsis -> snowman renaming.

File: e2e/tslint.json [hunk 4]
```diff
@@ -46,6 +46,9 @@
     "no-consecutive-blank-lines": [
       true,
       2
+    ],
+    "triple-equals": [
```
Comment (als364) [inline e2e/tslint.json:50]: Why are you disabling this instead of making the code conform with this rule?

File: e2e/app/component/card-base.ts (hunk 1 above)
Comment (aweng98) [reply e2e/app/component/card-base.ts:9]: Finding icon and clicking icon to trigger snowman menu to open is not in `snowman-menu` class because icons' xpath selectors are no same on all pages.

File: e2e/app/page/workspace-base.ts (hunk 2 above)
Comment (aweng98) [reply e2e/app/page/workspace-base.ts:230]: Icon is not standard ellipsis icon. xpath selector is different from standard three-dots.
Screen Shot 2020-09-30 at 12 15 00 PM

File: e2e/tests/conceptsets/conceptset-edit.spec.ts (hunk 3 above)
Comment (aweng98) [reply e2e/tests/conceptsets/conceptset-edit.spec.ts:28]: you're right. I can revert the change. I thought this test doesn't need a new workspace to run.

File: e2e/tslint.json (hunk 4 above)
Comment (aweng98) [reply e2e/tslint.json:50]: Because tslint fails without this option for this code https://github.com/all-of-us/workbench/blob/959d28198cd80fb3ac0c451b35bb6f4b2574fb51/e2e/tests/notebook/notebook-download.spec.ts#L18
From CH: the `==` is intentional, treat `null` as false.

File: e2e/app/page/workspace-base.ts (hunk 2 above)
Comment (als364) [reply e2e/app/page/workspace-base.ts:230]: Argh, of course it's a fontawesome icon here and a clarity icon everywhere else

File: e2e/app/component/card-base.ts (hunk 1 above)
Comment (als364) [reply e2e/app/component/card-base.ts:9]: Right, but for the most part, the `.//clr-icon[@shape="ellipsis-vertical"]` is used in all the xpath selectors to get the snowman icon. That text doesn't need to be duplicated everywhere.

File: e2e/tests/conceptsets/conceptset-edit.spec.ts (hunk 3 above)
Comment (als364) [reply e2e/tests/conceptsets/conceptset-edit.spec.ts:28]: Yes, please split this into a separate PR

File: e2e/tslint.json (hunk 4 above)
Comment (als364) [reply e2e/tslint.json:50]: Right, but why not do a null check instead? It's worth it to not loosen the enforcement around triple equals. Double equals really does do some strange and unintuitive things in Javascript and even if it's useful here it's harmful elsewhere.

File: e2e/app/page/workspace-base.ts (hunk 2 above)
Comment (als364) [reply e2e/app/page/workspace-base.ts:230]: An RW ticket should be created to change this icon into a clarity icon (or all Snowman icons into fontawesome icons); would you like to do that or would you like me to do that?

File: e2e/tests/conceptsets/conceptset-edit.spec.ts (hunk 3 above)
Comment (aweng98) [reply e2e/tests/conceptsets/conceptset-edit.spec.ts:28]: Done.

File: e2e/tslint.json (hunk 4 above)
Comment (aweng98) [reply e2e/tslint.json:50]: Okay. But on a second thought, I rather leave file unchanged in this PR. Change reverted.

File: e2e/app/page/workspace-base.ts (hunk 2 above)
Comment (aweng98) [reply e2e/app/page/workspace-base.ts:230]: Filed new ticket [RW-5682](https://precisionmedicineinitiative.atlassian.net/browse/RW-5682)

File: e2e/app/component/card-base.ts (hunk 1 above)
Comment (aweng98) [reply e2e/app/component/card-base.ts:9]: Where is a good place to put it as common str? Not so many places, I found only two files have the string.
Screen Shot 2020-09-30 at 3 45 54 PM

Comment (als364) [reply e2e/app/component/card-base.ts:9]: I'd put it in `snowman-menu.ts`, export the const, then access that from `card-base.ts` and `cohort-participants-group.ts`

File: e2e/app/page/workspace-base.ts (hunk 2 above)
Comment (als364) [reply e2e/app/page/workspace-base.ts:230]: thank you!

File: e2e/app/component/card-base.ts (hunk 1 above)
Comment (aweng98) [reply e2e/app/component/card-base.ts]: yes, good call. Done!

**Review by als364 — APPROVED** [review]
